Comparing samples with DESeq2
---------------------------------------

So far both us and other groups have been comparing samples using `bedtools`.
In other words, coordinates of peaks identified in each sample were overlapped,
and in case of an overlap a peak was considered to be common for two samples.

```
sample 1:          ------------
sample 2:                  ------------
sample 3:   ---- 

Overlap: 1 peak between sample 1 and sample 2.
```

This way we do not appreciate possible difference in the coverage of the peak,
in other words the intensity of the signal.

In order to compare peaks in a statistically valid way and discover peaks
present in multiple samples with different intensity, we decided to try
`DESeq2`. It is a statistical package for `R` initially developed to identify
differentially expressed genes. Essentially it needs to know the coverage of
certain regions in different samples, and it calculates the p-value of the
coverage being significantly different between sample groups. This is exactly
what we need -- we have defined regions (called peaks), we can count reads
mapped to each peak and we want to know which peaks have significantly
different coverage in different sample groups (conditions, e.g., fresh against
frozen cells).

Hereby we explored how `DESeq2` performs on ATAC data.

#### DESeq2 for peaks common for fresh and frozen samples

We started testing `DESeq2` of fresh and frozen K562 samples, as they seemed to
be of the best quality:

- fresh cells (3 replicates)
- frozen cells (3 replicates)

Firstly, we wanted to show that `DESeq2` can look at peaks common for two
conditions and pinpoint at the ones which have significantly different
coverage. Therefore we generated a list of peaks common between two conditions:

1) Using three input narrowPeak files generated by a peak caller, get union of
   peaks with `bedtools` *using the pipeline of Jason Hendry*. A peaks should
   have at least one base pair overlap across all three replicates to be
   considered a true peak:

```
replicate 1 ------------
replicate 2          ------------
replicate 3    -------
```

2) After we have generated a union peak list for fresh and frozen samples,
   generate a union of those peak lists with `bedtools` *using the pipeline of
   Jason Hendry*. This will generate the **final peak list**.

3) Using `samtools view` count reads mapped to each of the peaks from the final
   peak list in each of the replicates. This generates a matrix with rows being
   common peaks and columns being numbers of reads mapped to each peak in each
   replicate. This is the input file that we feed to `DESeq2`.


**Normalization**

Fresh and frozen samples have very different library sizes, and we want to make
sure that the normalization performed by `DESeq2` accounts for that. This
figure shows that the normalization indeed works and peaks which are not
supposed to be differentially expressed are scaled to the same level:


![alt text](https://github.com/jknightlab/ATACseq_pipeline/blob/master/Core_manuscript/DESeq2/fresh_frozen.raw_vs_norm.png)











