Primary quality control of K562 samples
---------------------------------------

### Analysis results

Hereby we explored 12 ATAC samples generated for different conditions of
chromatin for K562 cell line:

- fresh cells (3 replicates)
- frozen cells (3 replicates)
- fixed cells, day 3 after fixation (3 replicates)
- fixed cells, day 7 after fixaton (3 replicates)

The following measures were calculated by running `pinechrom` pipeline:

- Average fragment length
- Minimum fragment length
- Median fragment length
- Maximum fragment length
- Number of mapped reads after filtering
- Number of called peaks normalized by total mapped reads
- Number of called peaks normalized by reads mapped per chromosome
- Reads in peaks normalized by total mapped reads (to get an idea of
  coverage/peak height/signal intensity)
- Signal to noise ratio
- Average peak width
- Percentage of peaks common for replicates within one condition (fresh,
  frozen, etc) *taken from reports generated by Silvia Salatino*
- Percentage of peaks from each condition overlapping with DNase1 peaks of K562
  *taken from reports generated by Silvia Salatino*
- Amount of peaks common between different conditions *generated by Jason Hendry*

The plots were generated with
[these commands](https://github.com/jknightlab/ATACseq_pipeline/blob/master/Core_manuscript/reports_summary_plots.R)
from
[this file](https://github.com/jknightlab/ATACseq_pipeline/blob/master/Core_manuscript/reports_summary.txt).

Single PDF file containing all plots in high resolution can be found
[here](https://github.com/jknightlab/ATACseq_pipeline/blob/master/Core_manuscript/K562.three_repl.boxplot.QC.pdf).

### General observations and conclusions

**1. Low number of good-quality reads in fresh samples.**

We observe less than 5 mln good quality reads in each replicate of fresh
samples and above 10 mln good quality reads for each replicate of frozen/fixed
samples. As the amount of input material was the same in all experiments, we do
not expect such a big difference. The explanation is:

- input number of mapped reads was >80 mln for frozen and 44-54 mln for
  fresh/fixed;
- we lost >80% reads while removing duplicates and mitochondrial reads for
  fresh samples and 35-67% reads for frozen/fixed.

This is the first time we see such high rate of duplicates in any of previously
analyzed cell lines/samples from the same cell line. We do not know what caused
such high duplication rate. Three fresh K562 replicates from the same sample
will be re-sequenced on HiSeq 4000 to confirm this observation.


![alt text](https://github.com/jknightlab/ATACseq_pipeline/blob/master/Core_manuscript/K562_General_QC/K562.mapped_reads.png)


**2. Fragment size**

|          |           |
| -------- | --------- |
| ![alt text](https://github.com/jknightlab/ATACseq_pipeline/blob/master/Core_manuscript/K562_General_QC/K562.min_frag_len.png) | ![alt text](https://github.com/jknightlab/ATACseq_pipeline/blob/master/Core_manuscript/K562_General_QC/K562.median_frag_len.png) |
| ![alt text](https://github.com/jknightlab/ATACseq_pipeline/blob/master/Core_manuscript/K562_General_QC/K562.max_frag_len.png) | ![alt text](https://github.com/jknightlab/ATACseq_pipeline/blob/master/Core_manuscript/K562_General_QC/K562.average_frag_len.png) |


We can see that fragment size distribution is similar for fresh and frozen
samples; also, it looks similar to the typical ATAC fragment distribution
previously observed by others and us in other cell types. Fragment size
distribution for fixed_day3 has a more pronounced nucleosome pattern, however,
lacks a pick at the shortest fragment length present in fresh and frozen cells.
Fragment size distribution for fixed_day7 is very noisy and variable across
replicates for fragment sizes between 50 bp and 600 bp, which is not observed
for the other three conditions.

![alt text](https://github.com/jknightlab/ATACseq_pipeline/blob/master/Core_manuscript/fragment_size_comparison.png)

When we look at the distribution of fragment length put on the same Y axis and
compare it with the distribution of fragment sizes in Greenleaf dataset, we can
see that fresh sample does not have a sharp narrow peak around 50bp fragment
length (probably due to the longer read length -- 100 bp instead of 50 bp, as
in Greenleaf and frozen/fixed datasets). We also notice that the maximum
fragment length is similar in frozen and Greenleaf datasets and shorter  in
fresh dataset. Fixed cells do not produce a peak around 50bp, have a more
pronounced nucleosomal signal (deeper drops between the peaks at nucleosomal
sizes), contain longer fragments (longer than 1 kB). Overall, the profile
generated using frozen dataset is the closest to the profile of Greenleaf ATAC
data.

|       |       |
| ----- | ----- |
| Greenleaf ATAC dataset | ![alt text](https://github.com/jknightlab/ATACseq_pipeline/blob/master/Core_manuscript/fragm_size/Greenleaf_fragments.png) |

|       |       |
| ----- | ----- |
| ![alt text](https://github.com/jknightlab/ATACseq_pipeline/blob/master/Core_manuscript/fragm_size/fresh_fragm.png) | ![alt text](https://github.com/jknightlab/ATACseq_pipeline/blob/master/Core_manuscript/fragm_size/frozen_fragm.png) |r
| ![alt text](https://github.com/jknightlab/ATACseq_pipeline/blob/master/Core_manuscript/fragm_size/fixed3_fragm.png) | ![alt text](https://github.com/jknightlab/ATACseq_pipeline/blob/master/Core_manuscript/fragm_size/fixed7_fragm.png) |

As can be appriciated from the boxplots mentioned above, minimum fragment
length is lower and less variable in fresh than in frozen/fixed samples. Median
fragment length is the highest and most variable for fixed_day7 and the maximum
fragment length is the highest for fixed_day3 and fixed_day7. Such high
fragment length as 2000 bp has not been observed by us before when sequencing
fresh samples. This might be explained by the fact that fixation inactivates
all enzymes, including endonucleases which are still active in fresh samples
and might be degrading the fragments decreasing their maximum length. However,
freezing should also inactivate the enzymes in a similar way, but we do not see
such high fragment length for frozen samples. We can also see that regardless
of high maximum fragment length, the average fgrament length of fresh and
frozen sample (450-500 bp) is much higher than for fixed samples (320-370). We
can conclude that long fragments are not highly abundant in fixed samples.

**3. Peak analysis**

|     |     |
| --- | --- |
| ![alt text](https://github.com/jknightlab/ATACseq_pipeline/blob/master/Core_manuscript/K562_General_QC/K562.num_peaks.normalized.png) |  ![alt text](https://github.com/jknightlab/ATACseq_pipeline/blob/master/Core_manuscript/K562_General_QC/K562.num_peaks_per_chrom.png) |
| ![alt text](https://github.com/jknightlab/ATACseq_pipeline/blob/master/Core_manuscript/K562_General_QC/total_peaks_norm_total_reads.png) | ![alt text](https://github.com/jknightlab/ATACseq_pipeline/blob/master/Core_manuscript/K562_General_QC/K562.reads_in_peaks.png) |

As can be appreciated from the boxplots mentioned above, number of peaks called
in each condition (when normalized by the number of mapped reads) is the
highest in fresh samples and the least variable across replicates. The plot
"Reads in peaks normalized by total mapped reads" gives an idea about the
intensity of the signal, as it indirectly points at coverage/intensity of
identified peaks -- number of reads mapped to those peaks. We can see that
peaks identified in frozen samples are on average deeper covered than the peaks
identified in fresh, with a similar standard deviation. The intensity of peaks
in fixed samples is lower and more variable across the replicates.

|     |     |
| --- | --- |
| ![alt text](https://github.com/jknightlab/ATACseq_pipeline/blob/master/Core_manuscript/K562_General_QC/K562.signal_to_noise.png) | ![alt text](https://github.com/jknightlab/ATACseq_pipeline/blob/master/Core_manuscript/K562_General_QC/K562.peak_width.png) |

Signal to noise ratio is a measure introduced by us to access the quality of
the ATAC experiment -- it calculates the number of reads mapped *on target* (to
peaks) divided by the number of reads mapped *off target* (outside peaks, to
randomly selected regions of the same width as peaks). Signal-to-noise ratio of
**3** is something that we saw previously for a different cell line in
experiments generated by us and others. This implies that fresh and frozen
samples have a much stronger signal (on-target mapping) than the fixed ones.

If we look at the peak width, we can see that the average peak width is the
highest for frozen, reasonably high for fresh and the lowest for fixed samples
(~170 bp compared to ~230). Once again, that means that the long fragments
present in fixed samples do not contribute much.

**4. Reproducibility of peaks**

|     |     |
| --- | --- |
| ![alt text](https://github.com/jknightlab/ATACseq_pipeline/blob/master/Core_manuscript/K562_General_QC/K562.overlap_across_replicates.png) | ![alt text](https://github.com/jknightlab/ATACseq_pipeline/blob/master/Core_manuscript/K562_General_QC/K562.overlap_with_DNase.png) |


When we look at the number of peaks common for different replicates within one
condition, we can see that 70% of peaks in frozen are common for all three
replicates; and 60% of peaks in fresh are common for all three replicates.
However, when we look at fixed cells, only 30% of peaks are common in
fixed_day3 and only 10% of peaks are common in fixed_day7.

When we look at the number of peaks common between ATAC and publicly available
DNase data for K562, we can see that 80-90% of peaks are common between ATAC
and DNase for fresh and frozen; however, we observe only 10-40% overlap between
the peaks identified in DNase and fixed ATAC samples.

**5. Comparison of peaks under different conditions**

To compare different conditions, we created a summary peak list per condition
using `bedtools intersect` and only peaks present in all three replicates
were considered. The same peak in all three conditions has to have at least 1 bp
overlap. After that we compared two conditions: the summary peaks have to have a
certain overlap, 1bp or 50% of each peak length.


|          |           |
| -------- | --------- |
| Summary peaks had at least one base pair overlap | ![alt text](https://github.com/jknightlab/ATACseq_pipeline/blob/master/Core_manuscript/K562_General_QC/Macs2_venns-1bp.png) |

The overlap of peaks with minimum 50% overlap between two peaks can be found
[here](https://github.com/jknightlab/ATACseq_pipeline/blob/master/Core_manuscript/K562_General_QC/Macs2_venns-50per.png).


**General conclusion**

1. We observe that fresh, frozen and fixed samples produced an ATAC-like
fragment length distribution and reasonable number of good quality reads.
Average fragment length is much higher for fresh and frozen samples (450-500 bp
compared to 320-370 bp). However, we would like to address fresh samples once
more, as we expected more good quality reads from those samples.

2. Normalized number of peaks is the highest for fresh samples (comparable for
frozen). Intensity of peaks (number of reads mapped to peaks) and average peak
width is the highest for frozen samples (comparable for fresh). Fixed samples
show less peaks with lower intensity and higher variability across replicates.

3. Peaks in fresh and frozen samples are much more reproducible than peaks in
fixed samples (>60% against <40%). Peaks in frozen samples are a bit more
consistent when comparing three replicates; peaks in fresh samples are a bit
more consistent when comparing with DNase samples. However, in this respect the
difference between the results generated from fresh and frozen samples are
minor.

4. Signal-to-noise ratio shows that fresh and frozen samples have a very
similar signal strength and are much cleaner than fixed samples, which look
much noisier.

5. Overall, it looks like fresh and frozen are worth further exploration, while
fixed samples provide a very noisy, not reproducible signal.




--------------------
#### Developed by Irina Pulyakhina irina@well.ox.ac.uk
