Primary quality control of K562 samples
---------------------------------------

#### Analysis results

Hereby we explored 12 ATAC samples generated for different conditions of
chromatin for K562 cell line:

- fresh cells (3 replicates)
- frozen cells (3 replicates)
- fixed cells, day 3 after fixation (3 replicates)
- fixed cells, day 7 after fixaton (3 replicates)

The following measures were calculated by running `pinechrom` pipeline:

- Average fragment length
- Minimum fragment length
- Median fragment length
- Maximum fragment length
- Number of mapped reads after filtering
- Number of called peaks normalized by total mapped reads
- Number of called peaks normalized by reads mapped per chromosome
- Reads in peaks normalized by total mapped reads (to get an idea of
  coverage/peak height/signal intensity)
- Signal to noise ratio
- Average peak width
- Percentage of peaks common for replicates within one condition (fresh,
  frozen, etc) *taken from reports generated by Silvia Salatino*
- Percentage of peaks from each condition overlapping with DNase1 peaks of K562
  *taken from reports generated by Silvia Salatino*

The plots were generated with
[these commands](https://github.com/jknightlab/ATACseq_pipeline/blob/master/Core_manuscript/reports_summary_plots.R)
from
[this file](https://github.com/jknightlab/ATACseq_pipeline/blob/master/Core_manuscript/reports_summary.txt).

Single PDF file containing all plots in high resolution can be found
[here](https://github.com/jknightlab/ATACseq_pipeline/blob/master/Core_manuscript/K562.three_repl.boxplot.QC.pdf).

#### General observations and conclusions

**1. Low number of good-quality reads in fresh samples.**

We observe less than 5 mln good quality reads in each replicate of fresh
samples and above 10 mln good quality reads for each replicate of frozen/fixed
samples. As the amount of input material was the same in all experiments, we do
not expect such a big difference. The explanation is:

- input number of mapped reads was >80 mln for frozen and 44-54 mln for
  fresh/fixed;
- we lost >80% reads while removing duplicates and mitochondrial reads for
  fresh samples and 35-67% reads for frozen/fixed.

This is the first time we see such high rate of duplicates in any of previously
analyzed cell lines/samples from the same cell line. We do not know what caused
such high duplication rate. Three fresh K562 replicates from the same sample
will be re-sequenced on HiSeq 4000 to confirm this observation.

**2. Fragment size**

We can see that fragment size distribution is similar for fresh and frozen
samples; also, it looks similar to the typical ATAC fragment distribution
previously observed by others and us in other cell types. Fragment size
distribution for fixed_day3 has a more pronounced nucleosome pattern, however,
lacks a pick at the shortest fragment length present in fresh and frozen cells.
Fragment size distribution for fixed_day7 is very noisy and variable across
replicates for fragment sizes between 50 bp and 600 bp, which is not observed
for the other three conditions.

![alt text](https://github.com/jknightlab/ATACseq_pipeline/blob/master/Core_manuscript/fragment_size_comparison.png)

As can be appriciated from the boxplots mentioned above, minimum fragment
length is lower and less variable in fresh than in frozen/fixed samples. Median
fragment length is the highest and most variable for fixed_day7 and the maximum
fragment length is the highest for fixed_day3 and fixed_day7. Such high
fragment length as 2000 bp has not been observed by us before when sequencing
fresh samples. This might be explained by the fact that fixation inactivates
all enzymes, including endonucleases which are still active in fresh samples
and might be degrading the fragments decreasing their maximum length. However,
freezing should also inactivate the enzymes in a similar way, but we do not see
such high fragment length for frozen samples. We can also see that regardless
of high maximum fragment length, the average fgrament length of fresh and
frozen sample (450-500 bp) is much higher than for fixed samples (320-370). We
can conclude that long fragments are not highly abundant in fixed samples.






