## Functional Genomics of Ankylosing Spondylitis

### Analyzing RNA-Seq data

We are sequencing mRNA and miRNA of a number of different cell types from
several patients to investigate the differences between cell types, cell types
most relevant for the disease and further integrate RNA-Seq with ATAC-Seq and
CaptureC. This page contains some guidelines to analyze RNA-Seq data.

Such standard procedures as adapter trimming, mapping to the human genome/miRNA
database are done by the Core. We need to perform some quality checks on the
bam files, remove duplicates, generate count files containing gene expression
per sample, run differential expression analysis.

**1. Quality checks and filtering of bam files**

We need to make sure that bam files generated by the Core are of good quality.
For that, we count number of total sequenced reads, number of mapped reads,
number of mapped reads after removing duplicates, number of uniquely mapped
reads and number of reads mapped to known genes. We need to write a script that
would automatically do that.

**2. Differential gene expression**

To get the list of differentially expressed genes, we first need to calculate
gene expression -- generate a matrix text file containing number of reads
mapped to each gene in each sample. We will further need to group the samples
based on the cell type/phenotype and run differential expression analysis with
DESeq. This analysis will be run pairwise for each two groups of sampes.  DESeq
will report all genes expressed in our cell types, their expression (normalized
by the library size) and p-values reflecting whether this gene is
differentially expressed in any cell types.  We need to write a script that
would run DESeq and report some statistics (number of expressed genes, number
of differentially expressed genes at different p-value cut-offs, etc).

**3. Pathway analysis**

After discovering differentially expressed genes, we need to classify them
based on their function and run pathway analysis. As we will have to run
pathway analysis multiple times, ideally we need to find a pathway analysis
software that can be run from the command-line.

**4. Known genes relevant for AS**

When we have the results of differential gene expression and a list of
differentially expressed genes, we should look for genes from other studies in
human or model organisms that were identified as relevant for AS (nt
necessarily the same cell type). We would compare previously identified genes
with the genes that we define as differentially expressed/relevant for AS.

We should also look for any known miRNAs or miRNA targets to be better
interpret our miRNA data.

**5. Integrating mRNA and miRNA data**

As we will have both mRNA and miRNA from the same patients and the same cell
types, we should find a way to integrate the two dataset to potentially explain
certain changes in gene expression using miRNA data.

**6. Integrating ATAC/DNase data for AS**

If we already have ATAC data on AS or can find publicly available DNase/Hi-C
data for AS, we should find a way to integrate that with our RNA data and try
to explain certain changes in gene expression using chromatin data.

**7. Integrate GWAS and our data**

Integrate M.Brown GWAS with our RNA-Seq and ATAC-Seq.



#### developed by Irina Pulyakhina irina@well.ox.ac.uk

