Signal to noise ratio in Chip-Seq
---------------------------------

In order to be able to compare signal-to-noise (**S2N**) ratio in ATAC to some
kind of standard, we decided to calculate S2N in Chip-Seq of K562. We
downloaded two technical
[replicates](https://www.encodeproject.org/experiments/ENCSR648INT/) generated
by ENCODE and, using ENCODE-generated bam files and bed files calculated S2N
using `pinechrom_general`.

### Data

We used:

| Replicate 1        | Replicate 2        |
| ------------------ | ------------------ |
| ENCFF676IGA.bam    | ENCFF234PKC.bam    |
| ENCFF237JZC.bed    | ENCFF207HZD.bed    |
| ENCFF334GZR.bigWig | ENCFF576SVU.bigWig |

#### Results

When we look at signal to noise ratio in this particular ChIP-Seq compared to
ATAC on, e.g., frozen K562 generated by the Core, we can see that signal to
noise ratio in ATAC is much higher. Note that the variation of signal to noise
ratio across chromosomes in Chip-Seq is much less than in ATAC.

![alt text](https://github.com/jknightlab/ATACseq_pipeline/blob/master/Core_manuscript/ChipSeq/s2n_boxplot.png)

Even though we expect ChIP-Seq to be very specific and less noisy, we can see
from this picture that both ChIP-Seq replicates are quite noisy and definitely
noisier compared to ATAC:

![alt text](https://github.com/jknightlab/ATACseq_pipeline/blob/master/Core_manuscript/ChipSeq/K562_chip_atac_noise.png)

#### Conclusion

Probably these ChIP-Seq samples are too noisy, noiser than typically, and we
would have to investigate more samples to get a better estimate of signal to
noise ratio in ChIP-Seq. However, these preliminary results are very promissing
and suggest that we are able to generate a very clean ATAC sample enriched for
the targeted signal.

---------------
Designed by Irina Pulyakhina, irina@well.ox.ac.uk

