ATAC-Seq pipeline
--------------------------------------

This page describes differential peak calling in ATAC-Seq data
using `macs2`.


### Overview of the program

The goal is to identify differential regions (regions showing different
intensity/coverage) betwee two conditions. This will be done using fuction
`bdgdiff` from ``macs2` toolkit. Note that `macs2 diffpeak` is not
implemented yet.

Peak calling using `macs2 callpeaks` should have been run prior to this
analysis. Among the results files generated by `macs2 callpeks` we are
interested in two files -- the file containing the background signal
(*control_lambda.bdg*) and the file containing the signal of the actual
sample (*treat_pileup.bdg*).

The analysis described here was performed on two skin samples, psoriatic
lesional and non-lesional skin. Standard ATAC protocol was followed by
sequencing on Illumina MiSeq.

### Parameters of *macs2 bdgdiff*

The following parameters were considered for examination:

-  `--cutoff`, logLR cutoff. This is the log likelyhood ratio and is 2 by
   default, which means that likelihood ratio equals 1000. This parameter was
nt changed.
- `--min-len`, minimum length of differential region. This parameter was set to
  10, 50, 75, 100, 125, 150, 175, 200.
- `--max-gap`, maximum gap to merge nearby differential regions. This parameter
  was not changed. (default is 100).
- `--depth1` and `--depth2`, sequencing depth of the two samples. These two
  parameters are used to calculate scaling factor for each sample and normalize
  the data. First, we used the default `--depth1 1` and `--depth2 1`. Then, in
  order to see the effect of normalization/scaling, based on the number of reads
  mapped to peaks we set `--depth1 790330` (number of reads mapped to peaks in
  lesional skin) and `--depth2 943409` (number of reads mapped to peaks in
  non-lesional skip).



