ATAC-Seq pipeline
--------------------------------------

This page describes differential peak calling in ATAC-Seq data
using `macs2`.


### Overview of the program

The goal is to identify differential regions (regions showing different
intensity/coverage) betwee two conditions. This will be done using fuction
`bdgdiff` from ``macs2` toolkit. Note that `macs2 diffpeak` is not
implemented yet.

Peak calling using `macs2 callpeaks` should have been run prior to this
analysis. Among the results files generated by `macs2 callpeks` we are
interested in two files -- the file containing the background signal
(*control_lambda.bdg*) and the file containing the signal of the actual
sample (*treat_pileup.bdg*).

The analysis described here was performed on two skin samples, psoriatic
lesional and non-lesional skin; and two monocyte samples, a publicly available
from Greenleaf lab and generated in-house. Standard ATAC protocol was followed by
sequencing on Illumina MiSeq for all in-house generated samples.

### Parameters of *macs2 bdgdiff*

The following parameters were considered for examination:

-  `--cutoff`, logLR cutoff. This is the log likelyhood ratio and is 2 by
   default, which means that likelihood ratio equals 1000. This parameter was
nt changed.
- `--min-len`, minimum length of differential region. This parameter was set to
  10, 50, 75, 100, 125, 150, 175, 200.
- `--max-gap`, maximum gap to merge nearby differential regions. This parameter
  was not changed. (default is 100).
- `--depth1` and `--depth2`, sequencing depth of the two samples. These two
  parameters are used to calculate scaling factor for each sample and normalize
  the data. First, we used the default `--depth1 1` and `--depth2 1`. Then, in
  order to see the effect of normalization/scaling, based on the number of reads
  mapped to peaks we set `--depth1 790330` (number of reads mapped to peaks in
  lesional skin) and `--depth2 943409` (number of reads mapped to peaks in
  non-lesional skip).

## Part 1: changing minimal length of overlap

### Results: statistics

In the table below the overlap (number of common peaks) between jknight monocytes
and lesional skin / greenleaf monocytes is shown when using different minimal required
overlap between two peaks to be selected. The column names stand for:

- GM -- monocytes generated in Greenleaf lab, not scaled/normalized to the number of reads
  in peaks
- GMS -- monocytes generated in Greenleaf lab, scaled to the number of reads in peaks
- GMS_% -- percent of peaks in overlap from the total number of peaks detected for
  the Greenleaf monocytes
- LS -- lesional skin, not scaled/normalized to the number of reads
  in peaks
- GMS -- lesional skin, scaled to the number of reads in peaks
- GMS_% -- percent of peaks in overlap from the total number of peaks detected for
  lesional skin


| minlength  | GM   | GMS  | GMS_% | LS  | LSS | LSS_% |
| ---------- | ---- | ---- | ----- | --- | --- | ----- |
| minlen 10  | 3137 | 2691 | 7.32  | 474 | 372 | 1.03  |
| minlen 50  | 2918 | 2519 | 6.85  | 380 | 323 | 0.89  |
| minlen 75  | 2735 | 2368 | 6.44  | 319 | 286 | 0.79  |
| minlen 100 | 2552 | 2212 | 6.02  | 259 | 260 | 0.72  |
| minlen 125 | 2370 | 2041 | 5.55  | 216 | 221 | 0.61  |
| minlen 150 | 2153 | 1838 | 5.00  | 176 | 182 | 0.50  |
| minlen 175 | 1904 | 1609 | 4.38  | 139 | 157 | 0.43  |
| minlen 200 | 1585 | 1353 | 3.68  | 99  | 124 | 0.34  |

### Results: examples

A [UCSC session](https://genome-euro.ucsc.edu/cgi-bin/hgTracks?hgS_doOtherUser=submit&hgS_otherUserName=pulyakhina&hgS_otherUserSessionName=macs2_diff_initial)
containint coverage files of Greenleaf monocytes, Jknight monocytes
and lesional skin data; bed files with common peaks for Greenleaf
monocytes vs Jknight monocytes and lesional skin vs Jknight monocytes
(with minimal required overlap of 100 bp).

Here are some examples of peaks detected as overlapping peaks:

Different width of overlap:

![alt text](https://github.com/jknightlab/ATACseq_pipeline/blob/master/macs2_diff/nrd1_greenleaf_not_skin.png)

Width and background noise matter more than the height of the peak:

![alt text](https://github.com/jknightlab/ATACseq_pipeline/blob/master/macs2_diff/two_peaks.png)


### Conclusions (*very* primary)

1. We have to scale the data, but we are not sure whether the current
scaling (inplemented in macs2) is the best.
2. A rather limited numbers of peaks is expected in the list of overlapping
peaks (around 10% rather than 90% for the same cell types generated in different
labs).
3. Both height, width and the background noise contribute to the decision whether
a peak will be reported as overlap.


## Part 2: changing likelihood


### Results: number of overlapping peaks with different likelihood ratios

This table contains number of peaks overlapping between monocyte data
generated at Jknight lab and at Greenleaf lab (hence ``GM''. Scaling was used to
make samples comparable (hence ``S''). Column 4 contains percent of total
peaks identified in Greenleaf monocytes that were in overlap with Jknight
monocyte peaks (hence ``%'').

| likelihood | minlength  | GMS  | GMS_% |
| ---------- | ---------- | ---- | ----- |
| 1          | minlen 75  | 8082 | 21.98 |
| 1          | minlen 100 | 7725 | 21.01 |
| 1          | minlen 125 | 7334 | 19.95 |
| 1          | minlen 150 | 6901 | 18.77 |
|            |            |      |       |
| 1.5        | minlen 75  | 5489 | 14.93 |
| 1.5        | minlen 100 | 5210 | 14.17 |
| 1.5        | minlen 125 | 4904 | 13.34 |
| 1.5        | minlen 150 | 4553 | 12.38 |
|            |            |      |       |
| 2          | minlen 75  | 3921 | 10.66 |
| 2          | minlen 100 | 3708 | 10.08 |
| 2          | minlen 125 | 3469 |  9.43 |
| 2          | minlen 150 | 3224 |  8.77 |
|            |            |      |       |
| 2.5        | minlen 75  | 3030 |  8.24 |
| 2.5        | minlen 100 | 2870 |  7.80 |
| 2.5        | minlen 125 | 2663 |  7.24 |
| 2.5        | minlen 150 | 2420 |  6.58 |

We can see from the table that even with mild cutoff for the likelihood
ratio (e.g., **1**) we can see only 20% peaks in overlap between the
same cell line (monocyte CD14+).

### Results: examples

This figure shows peaks identified as common or condition-specific using
the most relaxed cutoffs for both minlength and likelihood ratio. Still,
we can see that the results are very restricted.

Red box on the left indicates a peak which should have been identified as common but were not (probably due to lower signal in Jknight monocytes). Red box on the right indicates a peak which should have been identified as specific for Jknight monocytes.

![alt text](https://github.com/jknightlab/ATACseq_pipeline/blob/master/macs2_diff/example_imperfections_macs2.png)

### Intermediate conclusion

`macs2 bdgdiff` might not be the best tool to identify common and
condition-specific peaks in ATAC-Seq data. We will look for a better
tool to do that.


### Future tasks

1. Find out the best way of scaling/normalizing the data.
2. The remaining comparisons (vs non-lesional; Greenleaf vs lesional skin).
3. Look for good examples (skin-specific peaks, "house-keeping", monocyte-specific
peaks, stable peaks).
4. Find out the best statistical threshold (perhaps we should take a lower threshold
for likelyhood ratio, as the current likelyhood ratio was designed for ChIP-Seq peaks,
which are expected to generate cleaner signal).
5. Find not only common, but also condition-specific peaks.

