Comparing the performance of peak callers
------------------------------------------

No research on peak callers and their parameters for ATAC data has currently
been done. We decided to test two widely used peak callers -- Macs2 (initially
designed for ChIP-Seq data) and FSeq (initially designed for DNaseI data). We
ran peak calling using each of the tools and ranging two parameters which are
expected to have a major influence on peak calling:

- length/width/shifting of the statistical model which directly influences the
  average width of peaks to be called (`-l` for FSeq; `--extsize` for Macs2)
- significance threshold which determines how different from a background noise
  the signal should be to be considered as a peak (`-t` for FSeq; `--qvalue`
  for Macs2)

We used data generated from fresh K562 cells, as we consider it gold standard,
least biased (by freezing/fixing) and most accurate, therefore a perfect
candidate for testing peak callers on.

[This paper](http://journals.plos.org/plosone/article/asset?id=10.1371%2Fjournal.pone.0096303.PDF)
was used as a reference point as it performed a similar task for DNaseI data.

#### Experimental design

We had three samples (`bam` files as input for Macs2 and `bed` files as input
for FSeq) of fresh K562 cells, namely `fresh1`, `fresh2`, `fresh3`, which were
technical replicates of the same sample.

We ran
[FSeq](https://raw.githubusercontent.com/jknightlab/ATACseq_pipeline/master/Core_manuscript/Fseq_Macs2_Hotspot/run_fseq.sh)
ranging the two parameters in the following way:

- Paper: 0.001, 0.005, 0.05, 0.1, 0.5, 1, 1.5, 2, 2.5, 3, 3.5, 4, 4.5, 6.
- Us: 0.01, 1, 2, 4, 6, 8, 10, 12, 14, 16 (threshold)
- Us: 100 200 400 600 800 1000 2000 (length)

We ran
[Macs2](https://raw.githubusercontent.com/jknightlab/ATACseq_pipeline/master/Core_manuscript/Fseq_Macs2_Hotspot/run_macs.sh)
ranging the two parameters in the following way:

- Paper: 0.001, 0.005, 0.01, 0.02, 0.03, 0.04, 0.05, 0.06, 0.07, 0.08, 0.09, 0.1, 0.2, 0.3
- Us: 0.001, 0.005, 0.01, 0.02, 0.03, 0.05, 0.07, 0.09, 0.1, 0.5 (q-value)
- Us: 100 200 400 600 800 1000 2000 (extension)

Code to generate intersections (using Fseq peaks as example):
```
for l in `echo 100 200 400 600 800 1000 2000`
do
    for t in `echo 1.5 2 2.5 3 4 6 8 10 12 14 16`
    do
        fresh1=`find .. | grep "fresh1.filtered.bedpe.fseq.length_$l.tresh_$t\.na"`
        fresh2=`find .. | grep "fresh2.filtered.bedpe.fseq.length_$l.tresh_$t\.na"`
        fresh3=`find .. | grep "fresh3.filtered.bedpe.fseq.length_$l.tresh_$t\.na"`

        bases1=`cat $fresh1 | awk '{sum += $3-$2} END {print sum}'`
        bases2=`cat $fresh2 | awk '{sum += $3-$2} END {print sum}'`
        bases3=`cat $fresh3 | awk '{sum += $3-$2} END {print sum}'`

        overlap=`bedtools intersect -f 0.3 -r -a $fresh1 -b $fresh2 | \
            bedtools intersect -f 0.3 -r -a - -b $fresh3 | \
            bedtools sort -i - | \
            bedtools merge -i - | \
            awk '{sum += $3-$2} END {print sum}'`
        bedtools intersect -f 0.3 -r -a $fresh1 -b $fresh2 | \
            bedtools intersect -f 0.3 -r -a - -b $fresh3 | \
            bedtools sort -i - | \
            bedtools merge -i - > \
            fresh.fseq.l_$l.t_$t.bed

        echo length_$l.thresh_$t Overlap=$overlap Bases1=$bases1 Bases2=$bases2 Bases3=$bases3
    done
    echo
done
```


#### Choice of best parameters per peak caller

Choosing best parameters per peak caller turned out to be very complicated. We
looked at things like sensitivity, specificity, F-Score, distribution of called
peaks across annotated categories (TSS, enhancer regions, repressed regions,
etc). The problem with all those metrics is that when we increase the
width/length parameter, peak callers tend to call wider peaks, often join
narrow adjacent peaks. When we have wider window (2000 bases compared to 100),
we will most probably call more "true" events (annotated peaks), simply because
the covered area is so wide. Therefore we decided to look at a number of
metrics (FDR, sensitivity, specificity, F-Score) and choose the best parameter
set based on the combination of these metrics.

To identify "true annotated peaks", we need a good annotation which contains
only regions that we expect ATAC peaks to map to (such as promoters and
enhancers; but not repressed or transcribed regions). We looked at different
annotations and their combinations:

- Encode K562 regulatory elements annotation generated by Segwey
- Encode K562 regulatory elements annotation generated by ChromM
- Duke K562 DNaseI sites
- overlap of "1_active_promoter", "2_weak_promoter", "4_strong_enhancer",
  "6_weak_enhancer" from ChromM
- overlap of that with DNase
- overlap of "TSS", "Enhancer", "Weak_Enhancer" and "CTCF" from Segwey
- overlap of that with DNase
- overlap of "on target" areas (described above) between ChromM and Segwey
- overlap of "on target" areas (described above) between ChromM, Segwey and
  DNase

We came to the conclusion that Segwey "on target" areas are the best annotation
set. Annotation provided by ChromM contains too wide peaks (kbs wide).
Annotation of DNase peaks contains too narrow peaks (150 bp wide). Overlaps of
the annotations produce very limited, restricted lists of regions which do not
tend to be informative.

We used the intersection of peaks called in 3 fresh ATAC K562 samples (peaks
were called intersected if they had at least 30% overlap reciprocally in all
three samples). We required the overlap of 20% of ATAC peak length with an
annotated peak.

Code to generate intersections of peaks and the overlaps of these
intersections with the annotation:


- on_target="/well/bsgjknight/Irina_analysis/Annotation/segmentation.on_target.bed"
- off_target="/well/bsgjknight/Irina_analysis/Annotation/segmentation.off_target.bed"

```
for i in `ls fresh*bed`
do
    bases=`cat $i | awk '{sum += $3-$2} END {print sum}'`
    on_target=`bedtools intersect -f 0.2 -r \
        -a $i -b /well/bsgjknight/Irina_analysis/Annotation/seg.CTCF_TSS_E_WE.bed | \
        bedtools intersect -u -a - -b $i | \
        bedtools sort -i - | \
        bedtools merge -i - | \
        awk '{sum += $3-$2} END {print sum}'`
    off_target=`bedtools intersect -f 0.2 -r \
        -a $i -b /well/bsgjknight/Irina_analysis/Annotation/segmentation.off_target.bed | \
        bedtools intersect -u -a - -b $i | \
        bedtools sort -i - | \
        bedtools merge -i - | \
        awk '{sum += $3-$2} END {print sum}'`
    on_bases=`cat /well/bsgjknight/Irina_analysis/Annotation/seg.CTCF_TSS_E_WE.bed | \
        awk '{sum += $3-$2} END {print sum}'`
    off_bases=`cat /well/bsgjknight/Irina_analysis/Annotation/segmentation.off_target.bed | \
        awk '{sum += $3-$2} END {print sum}'`
    echo -e "$i\t$bases\t$on_bases\t$off_bases\t$on_target\t$off_target"
done | \
sed s/t_1.b/t_01.b/g | \
sed s/t_1.5.b/t_01.5.b/g | \
sed s/t_2.b/t_02.b/g | \
sed s/t_2.5.b/t_02.5.b/g | \
sed s/t_3.b/t_03.b/g | \
sed s/t_4.b/t_04.b/g | \
sed s/t_6/t_06/g | \
sed s/t_8/t_08/g | \
sed s/l_100.t/l_0100.t/g | \
sed s/l_200.t/l_0200.t/g | \
sed s/l_400.t/l_0400.t/g | \
sed s/l_600.t/l_0600.t/g | \
sed s/l_800.t/l_0800.t/g | \
sort -n -k 1
```


### Results

#### Choosing the best parameters for FSeq

Higher F-Score for length values 800, 1000 and 2000 (default 600 is not in this
list, indicating that the default settings might not be the ones showing the
best performance). The highest F-Score values were calculated for the
combinations of length=800 and threshold=2, length=2000 and threshold=4 and
length=1000 and threshold=2. We looked at some more stats for those parameter
combinations.


| Parameters               | F_Score | Overlap between replicates | %bases mapped to annotation | On_target / (On_target+Off_target) |
| ------------------------ | ------- | -------------------------- | --- | --- |
| length=800, threshold=2  | 0.527   | 15.33%                     | 70% | 97% |
| length=2000, threshold=4 | 0.529   | 46.8%                      | 69% | 97% |
| length=1000, threshold=2 | 0.533   | 17.2%                      | 66% | 96% |

Based on reasonably high F-Score, the highest specificity and the highet
percentage of bases mapped on target we chose the parameter set of
**length=800, threshold=2**.

It is worth mentioning that we replicated the results from the paper which
showed the same behaviour of F-Score with changing threshold values and
suggested to use a threshold value between 2 and 3 for DNaseI data.


#### Choosing the best parameters for FSeq

Higher Macs2 were observed for the extension values of 600 and 800 (default 200
is not in this list, indicating that the default settings might not be the ones
showing the best performance). The highest F-Score values were calculated for
the combinations of extension=800 and qvalue=0.05 and length=600 and
qvalue=0.07. Similarly to FSeq, here we looked at some more stats for those
parameter combinations.


| Parameters                 | F_Score | Overlap between replicates | %bases mapped to annotation | On_target / (On_target+Off_target) |
| -------------------------- | ------- | -------------------------- | --- | ----- |
| extension=800, qvalue=0.05 | 0.504   | 61.68%                     | 63% | 96.5% |
| extension=600, qvalue=0.07 | 0.525   | 46.8%                      | 69% | 97%   |

Based on reasonably high F-Score, the highest specificity and the highet
percentage of bases mapped on target we chose the parameter set of
**extension=600, qvalue=0.07**.

Again, in terms of qvalue we got very close to the results from the paper. They
suggest to use qval=0.05 for DNaseI data.


Based on the observations described above we chose the following parameters as
optimal:

| Peak caller | Default parameters | Chosen parameters   |
| ----------- | ------------------ | ------------------- |
| FSeq        | l=600, t=4         | **l=800, t=2**     |
| Macs2       | ext=200, q=0.05    | **ext=600, q=0.07** |


Here is what F-Score looks like:

![alt text](https://github.com/jknightlab/ATACseq_pipeline/blob/master/Core_manuscript/Fseq_Macs2_Hotspot/Fresh.Fseq_Macs2.Fscore.png)


F-Score, sensitivity and specificity, %bases mapped on target -- all that
information can be found:

- for FSeq [here](https://raw.githubusercontent.com/jknightlab/ATACseq_pipeline/master/Core_manuscript/Fseq_Macs2_Hotspot/fseq.bases_fscore.txt)
- for Macs2 [here](https://raw.githubusercontent.com/jknightlab/ATACseq_pipeline/master/Core_manuscript/Fseq_Macs2_Hotspot/macs.bases_fscore.txt)



#### Choosing the best peak caller

After we chose the best performing parameter set for both FSeq and Macs2, we
compared the two peak lists generated by each peak caller to see which peak
caller performs better. We looked at the following parameters:

- overlap between replicates
- overlap with various categories of the annotation
- signal-to-noise ratio


#### Overlap between replicates

We aimed to find out the best parameter to overlap replicates and select
reproducible peaks. When two peaks are overlapped, we requested at least 0.1%,
1%, 10%, 20%, ..., 100% of their length reciprocally to overlap (left panel of
the figure):

![alt text](https://github.com/jknightlab/ATACseq_pipeline/blob/master/Core_manuscript/Paper/Figures/Sup_Figure_XX.png)

We calculated the overlaps between three replicates (which you can see in the
table down below). After that, we plotted the difference between the two
adjacent values, in other words, how many peaks we gain when we relax the
cutoff for the minimum overlap. We chose the value of 20%, as for three out of
four peak callings it seemed to be within a plato of changes, before and after
that value the number of peaks increased rapidly.

**Overlap between replicates**

| Required overlap | Fseq default | Fseq chosen | Macs2 default | Macs2 chosen |
| ----- | ------ | ------ | ------ | ------ |
| 0.001 | 18.78% | 44.08% | 61.51% | 67.48% |
| 0.01  | 18.7%  | 43.44% | 61.21% | 67.28% |
| 0.1   | 16.76% | 36.34% | 58.26% | 65.57% |
| **0.2**   | **14.73%** | **29.09%** | **54.84%** | **63.61%** |
| 0.3   | 12.83% | 22.99% | 50.96% | 61.22% |
| 0.4   | 10.89% | 17.69% | 46.12% | 58.13% |
| 0.5   | 8.87%  | 13.06% | 39.27% | 53.22% |
| 0.6   | 6.65%  | 8.86%  | 30.23% | 45.72% |
| 0.7   | 4.32%  | 5.07%  | 19.67% | 35.03% |
| 0.8   | 2.02%  | 2.056% | 8.73%  | 20.82% |
| 0.9   | 0.36%  | 0.29%  | 1.69%  | 5.62%  |
| 1     | 0      | 0      | 0      | 0      |



#### Distribution of peaks across various categories of annotation

K562 segmentation track of annotated regulatory elements used by ENCODE was
taken from
[here](http://ucscbrowser.genap.ca/cgi-bin/hgTrackUi?hgsid=275287_Q3Wf4yyipacgB10HhqrM1BhHe10F&c=chr7&g=hub_1_genomeSegmentation).

This is the legend interpretation

- Distal CTCF/Candidate Insulator (CTCF)
- Candidate strong enhancer (E)
- Candidate weak enhancer (WE)
- Repressed (R)
- Promoter flanking (PF)
- Transcribed (T)
- TSS (TSS)


| Parameter                    | FSeq default     | FSeq chosen      | Macs2 default   | Macs2 chosen     |
| ---------------------------- | ---------------- | ---------------- | --------------- | ---------------- |
| Number of peaks              | 28,428           | 50,922           | 16,779          | 17,655           |
| Number of bases              | 15,841,550       | 32,970,365       | 8,525,614       | 20,401,332       |
| Overlap with transcribed     | 34,928 (0.2%)    | 275,438 (0.8%)   | 2,551 (0.03%)   | 68,539 (0.3%)    |
| Overlap with repressed       | 50,191 (0.3%)    | 282,679 (0.9%)   | 6,910 (0.03%)   | 71,053 (0.3%)    |
| Overlap with promoter flanks | 10,013 (0.1%)    | 33,752 (0.9%)    | 2,352 (0.08%)   | 5,065 (0.3%)     |
| Overlap with CTCF            | 604,345 (3.8%)   | 2,057,604 (6.2%) | 152,763 (1.8%)  | 406,491 (2%)     |
| Overlap with Weak enhancer   | 329,858 (2%)     | 879,292 (6.24%)  | 100,290 (1.2%)  | 157,732 (2%)     |
| Overlap with Enhancer        | 2,234,559 (14%)  | 5,190,366 (16%)  | 851,253 (10%)   | 2,945,415 (14%)  |
| Overlap with TSS             | 8,144,134 (51%)  | 13,286,438 (40%) | 4,338,659 (50%) | 11,855,778 (58%) |
| Overlap with "on target"     | 11,312,896 (71%) | 21,413,700 (64%) | 5,442,965 (63%) | 15,365,416 (75%) |
| Overlap with "off target"    | 95,132 (0.6%)    | 591,869 (1.8%)   | 11,813 (0.14%)  | 144,657 (0.7%)   |


![alt text](https://github.com/jknightlab/ATACseq_pipeline/blob/master/Core_manuscript/Fseq_Macs2_Hotspot/peak_distribution_across_annotation.png)

For the bottom two plots you can see what the distribution of peaks look like
when we group all "off target" categories (repressed and transcribed regions,
promoter flanking) and "on target" categories (enhancer, weak enhancer, TSS,
[CTCF](https://en.wikipedia.org/wiki/CTCF)).


#### Signal to noise ratio

This is how signal to noise ratio is calculated:

|         |                     |
|:-------:|:-------------------:|
| (1) To calculate signal to noise ratio, we first count number of reads mapped to the regions where peaks were called. For each peak, we also calculate its width and randomly select a region of an equal width on the areas of the genome where no peaks were called. | ![alt text](https://github.com/jknightlab/ATACseq_pipeline/blob/master/Core_manuscript/Paper/Figures/Sup_Figure_1.png) |
| (2) We then calculate the number of reads mapped to these regions. After that, we calculate a ratio between the number of reads mapped to peaks and the number of reads mapped to the selected regions outside peaks. | (3) We then normalize the ratio by the total number of reads in a sample used for peak calling and multiply the normalized value by 1,000,000 to scale it up. | 


And these are the values that we get:

| Parameter set            | Fresh K562 |
| ------------------------ | ---------- |
| FSeq default parameters  | 3.499      |
| FSeq chosen parameters   | 4.156      |
| Macs2 default parameters | 3.484      |
| Macs2 chosen parameters  | 6.79       |


# OLD

#### Illustrations

[This](https://genome-euro.ucsc.edu/cgi-bin/hgTracks?hgS_doOtherUser=submit&hgS_otherUserName=pulyakhina&hgS_otherUserSessionName=K562_fresh_FSeq_Macs2_chosen_parameters)
session was used to create the figures.

|             |            |
| ----------- | ---------- |
| ![alt text](https://github.com/jknightlab/ATACseq_pipeline/blob/master/Core_manuscript/Fseq_Macs2_Hotspot/UCSC_examples/one.png) | ![alt text](https://github.com/jknightlab/ATACseq_pipeline/blob/master/Core_manuscript/Fseq_Macs2_Hotspot/UCSC_examples/two.png)  |
| ![alt text](https://github.com/jknightlab/ATACseq_pipeline/blob/master/Core_manuscript/Fseq_Macs2_Hotspot/UCSC_examples/three.png) | ![alt text](https://github.com/jknightlab/ATACseq_pipeline/blob/master/Core_manuscript/Fseq_Macs2_Hotspot/UCSC_examples/four.png)  |
| ![alt text](https://github.com/jknightlab/ATACseq_pipeline/blob/master/Core_manuscript/Fseq_Macs2_Hotspot/UCSC_examples/five.png) | ![alt text](https://github.com/jknightlab/ATACseq_pipeline/blob/master/Core_manuscript/Fseq_Macs2_Hotspot/UCSC_examples/six.png)  |
| ![alt text](https://github.com/jknightlab/ATACseq_pipeline/blob/master/Core_manuscript/Fseq_Macs2_Hotspot/UCSC_examples/seven.png) | ![alt text](https://github.com/jknightlab/ATACseq_pipeline/blob/master/Core_manuscript/Fseq_Macs2_Hotspot/UCSC_examples/eight.png)  |


#### Conclusion

|                                          | FSeq      | Macs2     |
| ---------------------------------------- | --------- | --------- |
| overlap with "on target" annotation, %   | 66%       | 33%       |
| overlap with "off target" annotation, %  | 0.08%     | 0.02%     |
| overlap with "on target" annotation, bp  | 6,769     | 3,960     |
| overlap with "off target" annotation, pk | 9         | 3         |
| overlap with DNaseI peaks, bp            | 3,146,105 | 2,903,186 |
| overlap with DNaseI peaks, %             | 75%       | 89%       |
| peak width distribution                  | 380 bp    | 180 bp    |
| peak height distribution                 | higher    | lower     |
| overlap between replicates               | 67%       | 47%       |
| signal-to-noise ratio                    | 5.8       | 6.3       |

FSeq peaks are more enriched for the "on target" K562 regulatory elements; in
base pairs, they are more enriched for DNaseI peaks (but not in the % of
peaks). The average intensity of FSeq peaks is slightly higher and the overlap
between the replicates is much higher.

Macs2 peaks have a higher % overlap with DNaseI peaks, slightly higher
signal-to-noise ratio and narrower peaks.

Most importantly, when looking at FSeq and Macs2 peaks by eye, FSeq peaks look
more realistic.

Based on these observations we made a decision to use the following peak caller
to analyze our data further on: **FSeq, length 800, threshold 14**.

-------------------------------------------------
developed by Irina Pulyakhina irina@well.ox.ac.uk
